packages/cli/generators/app/templates/src/sequence.ts

```diff
@@ -0,0 +1,36 @@
+import {Context} from '@loopback/context';
+import {
+  DefaultSequence,
+  FindRoute,
+  InvokeMethod,
+  ParsedRequest,
+  ParseParams,
+  Reject,
+  Send,
+} from '@loopback/rest';
+import {ServerResponse} from 'http';
+
+export class MySequence extends DefaultSequence {
```

Comment:
We can't extend a Sequence because of DI ... we can't inject new sequence actions if we do this. Below is a sample implementation from `log-extension` I would suggest using (removing the logging stuff). 

```ts
    class LogSequence implements SequenceHandler {
      constructor(
        @inject(RestBindings.Http.CONTEXT) public ctx: Context,
        @inject(SequenceActions.FIND_ROUTE) protected findRoute: FindRoute,
        @inject(SequenceActions.PARSE_PARAMS)
        protected parseParams: ParseParams,
        @inject(SequenceActions.INVOKE_METHOD) protected invoke: InvokeMethod,
        @inject(SequenceActions.SEND) protected send: Send,
        @inject(SequenceActions.REJECT) protected reject: Reject,
        @inject(EXAMPLE_LOG_BINDINGS.LOG_ACTION) protected logger: LogFn,
      ) { }

      async handle(req: ParsedRequest, res: ServerResponse) {
        // tslint:disable-next-line:no-any
        let args: any = [];
        // tslint:disable-next-line:no-any
        let result: any;

        try {
          const route = this.findRoute(req);
          args = await this.parseParams(req, route);
          result = await this.invoke(route, args);
          this.send(res, result);
        } catch (err) {
          this.reject(res, req, err);
          result = err;
        }

        await this.logger(req, args, result);
      }
    }
```

Reply:
@virkt25 Are you saying `@inject` doesn't allow inheritance from base classes?

Reply:
Yes. I get the following error message
```
Uncaught Error: Cannot resolve injected arguments for function MySequence: The arguments[0] is not decorated for dependency injection, but a value is not supplied
```

When I try using the following MySequence (in log-extension acceptance test) ... when I'm trying to inject a custom sequence action.

```ts
class MySequence extends DefaultSequence {
      constructor(
        public ctx: Context,
        protected findRoute: FindRoute,
        protected parseParams: ParseParams,
        protected invoke: InvokeMethod,
        public send: Send,
        public reject: Reject,
        @inject(EXAMPLE_LOG_BINDINGS.LOG_ACTION) protected logger: LogFn,
      ) {
        super(ctx, findRoute, parseParams, invoke, send, reject);
      }

      // Handle your request routing here:
      async handle(req: ParsedRequest, res: ServerResponse) {
        let args: any = [];
        let result: any;

        try {
          const route = this.findRoute(req);
          args = await this.parseParams(req, route);
          result = await this.invoke(route, args);
          await this.send(res, result);
        } catch (err) {
          this.reject(res, req, err);
          result = err;
        }

        await this.logger(req, args, result);
      }
    }

    server.sequence(MySequence);
```

Reply:
I'm adding support for that in https://github.com/strongloop/loopback-next/pull/775.

Reply:
@raymondfeng I've made a commit with @virkt25 's suggestion in mind, but should I revert it if you're adding in support for inheritance in `@inject`?